### PR TITLE
Fixes related to transactions with username fields

### PR DESCRIPTION
**Pull request by andreibancioiu**
- On Node's API, added fields for sender & receiver username.
 - In the processing components, mark transactions with wrong username as invalid.

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by bogdan-rosianu — DISMISSED** [review]

File: node/node.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 	"sync/atomic"
 	"time"
 
-	"github.com/ElrondNetwork/elrond-go-logger"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
```
Comment (SebastianMarian) [inline node/node.go:16]: Not need for alias

File: process/transaction/shardProcess.go [hunk 2]
```diff
@@ -173,6 +173,13 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction) (vmco
 			}
 		}
 
+		if errors.Is(err, process.ErrUserNameDoesNotMatch) && txProc.flagRelayedTx.IsSet() {
+			receiptErr := txProc.executingFailedTransaction(tx, acntSnd, err)
+			if receiptErr != nil {
+				return vmcommon.UserError, receiptErr
```
Comment (SebastianMarian) [inline process/transaction/shardProcess.go:179]: For ErrInsufficientFunds error above, if executingFailedTransaction fails it is returned (0, receiptErr) and here (vmcommon.UserError, receiptErr), it is ok?

File: process/transaction/shardProcess.go [hunk 3]
```diff
@@ -173,6 +173,13 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction) (vmco
...
+			}
+		}
```
Comment (SebastianMarian) [inline process/transaction/shardProcess.go:181]: Should be retuned (vmcommon.UserError, err) or (vmcommon.UserError, nil) like in L188?

File: process/transaction/shardProcess.go (hunk 2 above)
Comment (andreibancioiu) [reply process/transaction/shardProcess.go:179]: CC: @sasurobert

Comment (sasurobert) [reply process/transaction/shardProcess.go:179]: Yes it is. User code return is not used on the return, only in tests.

File: process/transaction/shardProcess.go (hunk 3 above)
Comment (sasurobert) [reply process/transaction/shardProcess.go:181]: It is correct now. ReceiptErr must be returned as it signals ErrFailedTransaction

Comment (SebastianMarian) [reply process/transaction/shardProcess.go:181]: I want to say if it is missed a return between line 180-181: return vmcommon.UserError, nil, or it is correct to use the final return from line 190: return vmcommon.UserError, err ?

Comment (sasurobert) [reply process/transaction/shardProcess.go:181]: Yes. It is correct. That tx will not be taken into consideration when building the block.

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]

**Pull request merged**